flink-connector-mysql-cdc/src/test/java/com/alibaba/ververica/cdc/connectors/mysql/MySqlValidatorTest.java

```diff
@@ -0,0 +1,162 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.ververica.cdc.connectors.mysql;
+
+import org.apache.flink.api.connector.source.mocks.MockSplitEnumeratorContext;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.logical.utils.LogicalTypeUtils;
+
+import com.alibaba.ververica.cdc.connectors.mysql.source.MySqlParallelSource;
+import com.alibaba.ververica.cdc.connectors.mysql.source.utils.MySqlContainer;
+import com.alibaba.ververica.cdc.connectors.mysql.source.utils.UniqueDatabase;
+import com.alibaba.ververica.cdc.debezium.DebeziumSourceFunction;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.lifecycle.Startables;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static com.alibaba.ververica.cdc.connectors.mysql.MySqlTestUtils.basicSourceBuilder;
+import static com.alibaba.ververica.cdc.connectors.mysql.MySqlTestUtils.setupSource;
+import static java.nio.file.Paths.get;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/** Test for the {@link MySqlValidator}. */
+@RunWith(Parameterized.class)
+public class MySqlValidatorTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MySqlValidatorTest.class);
+
+    private static TemporaryFolder tempFolder;
+    private static File resourceFolder;
+    private static final RowType SPLIT_KEYS =
+            LogicalTypeUtils.toRowType(
+                    DataTypes.ROW(DataTypes.FIELD("id", DataTypes.BIGINT())).getLogicalType());
+    final boolean runInParallel;
+
+    @Parameterized.Parameters(name = "runInParallel = {0}")
```

Comment:
nit: use `runIncrementalSnapshot` as I discussed with jark recently, in fact the new Source supports single parallelism too, thus we plan to rename some option.